src/interfaces/statics.rs

```diff
@@ -401,139 +410,262 @@ mod tests {
+                    &mpk,
+                    &policy,
+                    "Department::MKG && Security Level::Low Secret",
+                    "low_secret_mkg_plaintext",
+                    Some(&header_metadata),
+                    None,
+                )?,
+
+                low_secret_fin_test_vector: EncryptionTestVector::new(
+                    &mpk,
+                    &policy,
+                    "Department::FIN && Security Level::Low Secret",
+                    "low_secret_fin_plaintext",
+                    None,
+                    None,
+                )?,
+            };
+            Ok(reg_vectors)
+        }
 
-        //let reg_vectors = NonRegressionTestVector {
-        //public_key: hex::encode(mpk.try_to_bytes()?),
-        //msk: hex::encode(msk.try_to_bytes()?),
-        //policy: hex::encode(serde_json::to_vec(&policy)?),
-        //user_decryption_key: hex::encode(top_secret_mkg_fin_user.try_to_bytes()?),
-        //user_decryption_key_2: hex::encode(medium_secret_mkg_user.try_to_bytes()?),
-        //encrypted_bytes: hex::encode(encrypted_bytes),
-        //plaintext: hex::encode(message),
-        //additional_data: hex::encode(additional_data),
-        //};
+        fn verify(&self) -> Result<(), Error> {
+            // top_secret_fin_key
+            self.low_secret_fin_test_vector
+                .decrypt(&self.top_secret_fin_key.key)?;
+            assert!(self
+                .low_secret_mkg_test_vector
+                .decrypt(&self.top_secret_fin_key.key)
+                .is_err());
+            assert!(self
+                .top_secret_mkg_test_vector
+                .decrypt(&self.top_secret_fin_key.key)
+                .is_err());
+
+            // top_secret_mkg_fin_key
+            self.low_secret_fin_test_vector
+                .decrypt(&self.top_secret_mkg_fin_key.key)?;
+            self.low_secret_mkg_test_vector
+                .decrypt(&self.top_secret_mkg_fin_key.key)?;
+            self.top_secret_mkg_test_vector
+                .decrypt(&self.top_secret_mkg_fin_key.key)?;
+
+            // medium_secret_mkg_key
+            assert!(self
+                .low_secret_fin_test_vector
+                .decrypt(&self.medium_secret_mkg_key.key)
+                .is_err());
+            self.low_secret_mkg_test_vector
+                .decrypt(&self.medium_secret_mkg_key.key)?;
+            assert!(self
+                .top_secret_mkg_test_vector
+                .decrypt(&self.medium_secret_mkg_key.key)
+                .is_err());
+            Ok(())
+        }
+    }
 
-        //std::fs::write(
-        //"non_regression_vector.json",
-        //serde_json::to_string(&reg_vectors).unwrap(),
-        //)
-        //.unwrap();
+    #[test]
+    fn test_generate_non_regression_vector() -> Result<(), Error> {
+        let reg_vector = NonRegressionTestVector::new()?;
+        std::fs::write(
+            // uncomment this to regenerate new test vector
```

Comment:
Why not to comment the call of this function instead ?

Reply:
I though we could eventually catch an encryption problem or key generation problem

Reply:
Then comment out the entire write, this will be clearer imho

Reply:
right, done